lib/client.js

```diff
@@ -114,7 +114,28 @@ Client.prototype._connect = function (callback) {
 
   function checkPgPass (cb) {
     return function (msg) {
-      if (self.password !== null) {
+      if (typeof(self.password) === 'function') {
+        try {
+          self._Promise.resolve(self.password())
+            .then(pass => {
+              if (undefined !== pass) {
+                if (typeof(pass) !== 'string') {
+                  con.emit('error', new TypeError('Password must be a string'))
+                  return
+                }
+                self.connectionParameters.password = self.password = pass
+              } else {
+                self.connectionParameters.password = self.password = null
+              }
+              cb(msg)
+            })
+            .catch(err => {
+              con.emit('error', err)
+            })
+        } catch (err) {
+          con.emit('error', err)
+        }
```

Comment:
wow - impressive error handling

Reply:
Apparently not as impressive as @charmander ;-)